configure.ac

```diff
@@ -951,6 +951,30 @@
         LIBNET_FAIL_WARN($libnet_dir)
     fi
 
+  # libbsd
+     AC_ARG_WITH(libbsd_includes,
+            [  --with-libbsd-includes=DIR  libbsd include directory],
+            [with_libbsd_includes="$withval"],[with_libbsd_includes=no])
+
+    if test "$with_libbsd_includes" != "no"; then
+        CPPFLAGS="${CPPFLAGS} -I${with_libbsd_includes}"
+    fi
+
+    LIBBSD=""
+    AC_CHECK_HEADER(bsd/sys/tree.h,, LIBBSD="no")
+
+    if test "$LIBBSD" = "no"; then
+        echo
+        echo "   ERROR!  libbsd-dev files not found, go get them"
+        echo
+        echo "   Ubuntu: apt-get install libbsd-dev"
+        echo "   Fedora: yum install libbsd-devel"
+        echo
+        exit 1
+    fi   
+     
+    AC_DEFINE([HAVE_SYS_TREE_H],[1],[bsd/sys/tree.h available])
+    
```

Comment:
bsd/sys/queue.h is a subset of sys/queue.h though the only difference is
the CIRCLEQ data structure.  Otherwise I believe it would be just a drop in
replacement.

On Sun, Oct 11, 2015 at 1:40 PM, Jason Ish <notifications@github.com> wrote:

> In configure.ac
> <https://github.com/inliniac/suricata/pull/1663#discussion_r41715101>:
>
> > +
> > +    LIBBSD=""
> > +    AC_CHECK_HEADER(bsd/sys/tree.h,, LIBBSD="no")
> > +
> > +    if test "$LIBBSD" = "no"; then
> > +        echo
> > +        echo "   ERROR!  libbsd-dev files not found, go get them"
> > +        echo
> > +        echo "   Ubuntu: apt-get install libbsd-dev"
> > +        echo "   Fedora: yum install libbsd-devel"
> > +        echo
> > +        exit 1
> > +    fi
> > +
> > +    AC_DEFINE([HAVE_SYS_TREE_H],[1],[bsd/sys/tree.h available])
> > +
>
> I wonder if we should just include tree.h with Suricata like we do
> queue.h... Or if depending on libbsd, we could drop queue.h and use the one
> from libbsd.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inliniac/suricata/pull/1663/files#r41715101>.
>